apps/dashboard/hooks/use-workspace-navigation.tsx

```diff
@@ -0,0 +1,17 @@
+import { useWorkspace } from "@/providers/workspace-provider";
+import { redirect } from "next/navigation";
+
+export const useWorkspaceNavigation = () => {
+  const { workspace, isLoading: isWorkspaceLoading } = useWorkspace();
+
+  if (isWorkspaceLoading) {
+    throw new Promise(() => {});
+  }
```

Comment:
Can we explain why we are doing this? I understand that we throw this then Suspense it, but flow is a bit hard to follow. Let's document it properly.